src/lib.rs

```diff
@@ -61,17 +65,43 @@ impl Environment {
     /// extern crate environment;
     ///
     /// use std::ffi::OsString;
+    /// use std::collections::HashMap;
     ///
     /// let e = environment::Environment::empty().insert("foo", "bar").compile();
-    /// assert_eq!(e, vec![(OsString::from("foo"), OsString::from("bar"))]);
+    ///
+    /// let mut hash_map: HashMap<OsString, OsString> = HashMap::new();
+    /// hash_map.insert("foo".into(), "bar".into());
+    ///
+    /// assert_eq!(e, hash_map);
     /// ```
     pub fn insert<S1: Into<OsString>, S2: Into<OsString>>(mut self, key: S1, val: S2) -> Self {
-        self.vars.push((key.into(), val.into()));
+        self.vars.insert(key.into(), val.into());
+        self
+    }
+
+    /// Remove an entry from the variables in this environment object
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// use std::ffi::OsString;
+    /// use std::collections::HashMap;
+    ///
+    /// let e = environment::Environment::empty().insert("foo", "bar").remove("foo").compile();
+    ///
+    /// let mut hash_map: HashMap<OsString, OsString> = HashMap::new();
+    ///
+    /// assert_eq!(e, hash_map);
+    /// ```
+    pub fn remove<S: Into<OsString>>(mut self, key: S) -> Self {
+        self.vars.remove(&key.into());
```

Comment:
This won't let you remove inherited variables.

Whether in this PR or another, we should probably move `::std::env::vars_os().chain(self.vars)` into `fn inherit`

Reply:
I was thinking about moving the var_os call at the instantiation. It avoid to have a `compile` method too.